core/src/main/java/org/elasticsearch/index/search/MatchQuery.java

```diff
@@ -87,8 +156,8 @@ public void setOccur(BooleanClause.Occur occur) {
         this.occur = occur;
     }
 
-    public void setCommonTermsCutoff(float cutoff) {
-        this.commonTermsCutoff = Float.valueOf(cutoff);
+    public void setCommonTermsCutoff(Float cutoff) {
```

Comment:
As far as I understand the code in MatchQuery#parse() it makes a difference whether this is set or not. With no cutoff defined, we get a BooleanQuery, with a value it's trying to build a ExtendedCommonTermsQuery. Not sure what a reasonable default would be, leaving this for now.